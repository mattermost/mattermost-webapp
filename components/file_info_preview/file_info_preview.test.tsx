// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import FileInfoPreview from 'components/file_info_preview/file_info_preview';

describe('components/FileInfoPreview', () => {
    test('should match snapshot, can download files', () => {
        const wrapper = shallow(
            <FileInfoPreview
                fileUrl='https://pre-release.mattermost.com/api/v4/files/rqir81f7a7ft8m6j6ej7g1txuo'
                fileInfo={{name: 'Test Image', size: 100, extension: 'jpg'}}
                canDownloadFiles={true}
            />,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, cannot download files', () => {
        const wrapper = shallow(
            <FileInfoPreview
                fileUrl='https://pre-release.mattermost.com/api/v4/files/aasf9afshaskj1asf91jasf0a0'
                fileInfo={{name: 'Test Image 2', size: 200, extension: 'png'}}
                canDownloadFiles={false}
            />,
        );

        expect(wrapper).toMatchSnapshot();
    });
});
