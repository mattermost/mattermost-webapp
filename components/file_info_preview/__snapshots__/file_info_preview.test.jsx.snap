// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FileInfoPreview should match snapshot, can download files 1`] = `
<div
  className="file-details__container"
>
  <a
    className="file-details__preview"
    href="https://pre-release.mattermost.com/api/v4/files/rqir81f7a7ft8m6j6ej7g1txuo"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      className="file-details__preview-helper"
    />
    <img
      src={null}
    />
  </a>
  <div
    className="file-details"
  >
    <div
      className="file-details__name"
    >
      Test Image
    </div>
    <div
      className="file-details__info"
    >
      File type JPG, Size 100B
    </div>
  </div>
</div>
`;

exports[`components/FileInfoPreview should match snapshot, cannot download files 1`] = `
<div
  className="file-details__container"
>
  <span
    className="file-details__preview"
  >
    <span
      className="file-details__preview-helper"
    />
    <img
      src={null}
    />
  </span>
  <div
    className="file-details"
  >
    <div
      className="file-details__name"
    >
      Test Image 2
    </div>
    <div
      className="file-details__info"
    >
      File type PNG, Size 200B
    </div>
  </div>
</div>
`;
